scripts/migrate_to_marathon_task_tag.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/hashicorp/consul/api"
+)
+
+type Config struct {
+	BootstrapAgentAddress string
+	ConsulTag             string
+}
+
+type MigrationStats struct {
+	MigratedServices                          int
+	SkippedFailedNodes                        int
+	SkippedFailedServices                     int
+	SkippedServicesNotManagedByMarathonConsul int
+	SkippedServicesAlreadyWithMarathonTaskTag int
+}
+
+func main() {
+	config := createConfig()
+	migrateSingleDC(config.BootstrapAgentAddress, config.ConsulTag)
+}
+
+func createConfig() *Config {
+	config := &Config{}
+	flag.StringVar(&config.BootstrapAgentAddress, "bootstrap-agent-address", "127.0.0.1:8500", "Address to one of the agents in the migrated DC")
+	flag.StringVar(&config.ConsulTag, "consul-tag", "marathon", "Tag specifying that a service is managed by marathon-consul")
+	flag.Parse()
+	return config
+}
+
+func migrateSingleDC(bootstrapAgentAddress string, consulTag string) *MigrationStats {
+	logrus.WithField("BootstrapAgentAddress", bootstrapAgentAddress).WithField("Tag", consulTag).Info("Starting migration...")
+
+	agentsPort, err := extractPort(bootstrapAgentAddress)
+	if err != nil {
+		logrus.WithError(err).WithField("BootstrapAgentAddress", bootstrapAgentAddress).
+			Error("Could not extract port from agent address, aborting.")
+		os.Exit(1)
+	}
+
+	client, err := api.NewClient(&api.Config{
+		Address: bootstrapAgentAddress,
+	})
+	if err != nil {
+		logrus.WithError(err).WithField("BootstrapAgentAddress", bootstrapAgentAddress).
+			Error("Could not create client to agent, aborting.")
+		os.Exit(1)
+	}
+
+	nodes, _, err := client.Catalog().Nodes(&api.QueryOptions{})
+	if err != nil {
+		logrus.WithError(err).Error("Could not fetch nodes, aborting.")
+		os.Exit(1)
+	}
+
+	logrus.Infof("Discovered %d node(s) to migrate", len(nodes))
+	stats := migrateNodes(nodes, agentsPort, consulTag)
+	logrus.WithField("Stats", stats).Info("Migration finished")
```

Comment:
I thought about it, but came to a conclusion that the effect would be the same, with much boilerplate code.